Libraries/LibWeb/CSS/Parser/PropertyParsing.cpp

```diff
@@ -4568,4 +4572,97 @@ RefPtr<CSSStyleValue const> Parser::parse_filter_value_list_value(TokenStream<Co
     return FilterValueListStyleValue::create(move(filter_value_list));
 }
 
+RefPtr<CSSStyleValue const> Parser::parse_contain_value(TokenStream<ComponentValue>& tokens)
+{
+    // none | strict | content | [ [size | inline-size] || layout || style || paint ]
+
+    // none
+    {
+        auto transaction = tokens.begin_transaction();
+        tokens.discard_whitespace();
+        auto keyword = parse_keyword_value(tokens);
+        if (keyword->to_keyword() == Keyword::None) {
+            if (tokens.has_next_token())
+                return {};
+            transaction.commit();
+            return keyword;
+        }
+    }
+
+    // strict
+    {
+        auto transaction = tokens.begin_transaction();
+        tokens.discard_whitespace();
+        auto keyword = parse_keyword_value(tokens);
+        if (keyword->to_keyword() == Keyword::Strict) {
+            if (tokens.has_next_token())
+                return {};
+            transaction.commit();
+            return keyword;
+        }
+    }
+
+    // content
+    {
+        auto transaction = tokens.begin_transaction();
+        tokens.discard_whitespace();
+        auto keyword = parse_keyword_value(tokens);
+        if (keyword->to_keyword() == Keyword::Content) {
+            if (tokens.has_next_token())
+                return {};
+            transaction.commit();
+            return keyword;
+        }
+    }
+
+    // [ [size | inline-size] || layout || style || paint ]
+    if (!tokens.has_next_token())
+        return {};
+
+    bool has_size = false;
+    bool has_layout = false;
+    bool has_style = false;
+    bool has_paint = false;
+
+    StyleValueVector containments;
+    while (tokens.has_next_token()) {
+        auto transaction = tokens.begin_transaction();
+        tokens.discard_whitespace();
+        auto keyword = parse_keyword_value(tokens);
+        switch (keyword->to_keyword()) {
```

Comment:
I don't think this is safe, since it assumes that `keyword` is always non-null at this point.

Reply:
Checking for null now.